test/explain/algorithm/test_gnn_explainer.py

```diff
@@ -0,0 +1,390 @@
+import pytest
+import torch
+import torch.nn.functional as F
+from torch.nn import Linear
+
+from torch_geometric.explain import Explainer
+from torch_geometric.explain.algorithm import GNNExplainer
+from torch_geometric.explain.config import (
+    ExplainerConfig,
+    MaskType,
+    ModelConfig,
+)
+from torch_geometric.nn import GATConv, GCNConv, global_add_pool
+
+
+# --------------------------------------------------------------------
+# model for node level tasks
+# --------------------------------------------------------------------
+class GCN_regression_multi_output(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.conv1 = GCNConv(3, 16)
+        self.conv2 = GCNConv(16, 7)
+
+    def forward(self, x, edge_index, batch=None):
+        x = self.conv1(x, edge_index)
+        x = F.relu(x)
+        return self.conv2(x, edge_index)
+
+
+class GCN_regression_single_output(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.conv1 = GCNConv(3, 16)
+        self.conv2 = GCNConv(16, 1)
+
+    def forward(self, x, edge_index, batch=None):
+        x = self.conv1(x, edge_index)
+        x = F.relu(x)
+        return self.conv2(x, edge_index)
+
+
+class GCN_classification_single_output(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.conv1 = GCNConv(3, 16)
+        self.conv2 = GCNConv(16, 7)
+
+    def forward(self, x, edge_index, batch=None):
+        x = self.conv1(x, edge_index)
+        x = F.relu(x)
+        x = self.conv2(x, edge_index)
+        return x.log_softmax(dim=1)
+
+
+class GCN_classification_multi_output(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.conv1 = GCNConv(3, 16)
+        self.conv2 = GCNConv(16, 7)
+        self.conv2_2 = GCNConv(16, 7)
+
+    def forward(self, x, edge_index, batch=None):
+        x = self.conv1(x, edge_index)
+        x = F.relu(x)
+        out_1 = self.conv2(x, edge_index).log_softmax(dim=1)
+        out_2 = self.conv2_2(x, edge_index).log_softmax(dim=1)
+        return torch.stack([out_1, out_2], dim=0)
+
+
+class GAT(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.conv1 = GATConv(3, 16, heads=2, concat=True)
+        self.conv2 = GATConv(2 * 16, 7, heads=2, concat=False)
+
+    def forward(self, x, edge_index, batch=None, **kwargs):
+        x = self.conv1(x, edge_index)
+        x = F.relu(x)
+        x = self.conv2(x, edge_index)
+        return x.log_softmax(dim=1)
+
+
+# --------------------------------------------------------------------
+# model for graph level tasks
+# --------------------------------------------------------------------
+
+
+class GNN_regression_multi_output(torch.nn.Module):
```

Comment:
I would rename to `GCN_graph_regression_multi_output`, and the node task version to `GCN_node_regression_multi_output`. Similarly for the other models.